python/pyarrow/_compute.pyx

```diff
@@ -967,6 +967,18 @@ class MakeStructOptions(_MakeStructOptions):
         self._set_options(field_names, field_nullability, field_metadata)
 
 
+cdef class _StructFieldOptions(FunctionOptions):
+    def _set_options(self, indices):
+        self.wrapped.reset(new CStructFieldOptions(indices))
+
+
+class StructFieldOptions(_StructFieldOptions):
+    def __init__(self, indices=None):
+        if indices is None:
+            indices = []
```

Comment:
For this specific case, do we actually want to allow not specifying any StructFieldOptions? (thus the default None here)

That would not select any field, which doesn't make much sense?

Reply:
Ah, I see in the tests that it then returns the original struct array, which is I suppose intended behaviour (also on the C++ side)

Reply:
It is, but that's not useful as a default behaviour, so the argument should still be mandatory IMHO.

Reply:
Ok, will set it to mandatory.